clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -578,10 +643,54 @@ func TestDeleteBasicScenarios(t *testing.T) {
 	}
 }
 
-func generateMachines() []*clusterv1.Machine {
+func generateMasterNode(name string) *clusterv1.Machine {
 	master := &clusterv1.Machine{}
-	master.Name = "test-master"
+	master.Name = name
 	master.Spec.Versions.ControlPlane = "1.10.1"
+	return master
+}
+
+func generateTestMasterMachine(masterNames []string) []*clusterv1.Machine {
+	var masters []*clusterv1.Machine
+	for _, mn := range masterNames {
+		m := generateMasterNode(mn)
+		masters = append(masters, m)
+	}
+	return masters
+}
+
+func generateTestNodeMachines(nodeNames []string) []*clusterv1.Machine {
+	var nodes []*clusterv1.Machine
+	for _, nn := range nodeNames {
+		n := &clusterv1.Machine{}
+		n.Name = nn
+		nodes = append(nodes, n)
+	}
+	return nodes
+}
+
+func generateInvalidExtractMasterMachine(masterNames, nodeNames []string) []*clusterv1.Machine {
+	masters := generateTestMasterMachine(masterNames)
+	nodes := generateTestNodeMachines(nodeNames)
+
+	var inputMachines []*clusterv1.Machine
+	inputMachines = append(inputMachines, masters...)
+	inputMachines = append(inputMachines, nodes...)
+	return inputMachines
+}
+
+func generateValidExtractMasterMachineInput(masterName string, nodeNames []string) []*clusterv1.Machine {
+	masters := generateMasterNode(masterName)
```

Comment:
Make masters an array (to match the plural name:

`masters = []*clusterv1.Machine{generateMasterNode(masterName)}`

Reply:
done